client/jetpack-onboarding/steps/contact-form.jsx

```diff
@@ -64,14 +100,48 @@ class JetpackOnboardingContactFormStep extends React.PureComponent {
 						}
 						image={ '/calypso/images/illustrations/contact-us.svg' }
 						onClick={ this.handleAddContactForm }
-						href={ getForwardUrl() }
+						href={ href }
 					/>
 				</TileGrid>
+			</Fragment>
+		);
+	}
+
+	render() {
+		const { basePath, getForwardUrl, hasContactForm, siteId, translate } = this.props;
+
+		return (
+			<div className="steps__main">
+				<DocumentHead title={ translate( 'Contact Form ‹ Jetpack Start' ) } />
+				<PageViewTracker
+					path={ [ basePath, STEPS.CONTACT_FORM, ':site' ].join( '/' ) }
+					title="Contact Form ‹ Jetpack Start"
+				/>
+				<QuerySites siteId={ siteId } />
+
+				<JetpackLogo full size={ 45 } />
+
+				{ hasContactForm ? (
+					<ConnectSuccess
+						href={ getForwardUrl() }
+						illustration="/calypso/images/illustrations/contact-us.svg"
+						onClick={ this.handleNextButton }
+						title={ translate( 'Success! Jetpack has added a "Contact us" page to your site.' ) }
+					/>
+				) : (
+					this.renderActionTile()
+				) }
 			</div>
 		);
 	}
 }
 
-export default connect( null, { saveJetpackOnboardingSettings } )(
-	localize( JetpackOnboardingContactFormStep )
-);
+export default connect(
+	( state, { settings, siteId, steps } ) => ( {
+		hasContactForm: !! get( settings, 'addContactForm' ),
+		isConnected: isJetpackSite( state, siteId ),
+		siteUrl: getUnconnectedSiteUrl( state, siteId ),
+		stepsPending: getJetpackOnboardingPendingSteps( state, siteId, steps ),
```

Comment:
Maybe it's time to move a few of these up to `main.jsx`, since we're using them in so many steps now.

Reply:
(Or maybe to a generic component for the steps that require a connection 🤔 )

Reply:
Yup, could be a good thing to do in a follow-up PR.